### ISSUE TITLE
Subtle daylight savings issue with Measure=Time, Timezone=X

### ISSUE THREAD

**Issue by dixonte** [rainmeter/rainmeter#322]
I've been using Rainmeter to function as a desktop world clock for a while. I'm using the TimeZone= property, and while the value I enter does seem to roughly correspond to the target timezone, I find that every time daylight savings begins or ends for *me* I have to update the value to get the correct time for my international friends. Then when daylight savings beings or ends for *them*, I need to do it again.

This doesn't seem right to me. I would expect to only have to update the value when daylight savings begins or ends in the target timezone, as only then should the offset relative to UTC change.

Rainmeter v4.5.16.3687 (64-bit), my timezone is Australian Eastern Standard Time (or Daylight Time, currently).

**Comment by brianferguson** [comment]
This is a common misconception between Daylight Savings Time and Timezones. While timezones are based purely on location, DST is a political decision made by governing bodies (like countries, states, or regions). In many places around the world, two locations within the same timezone can have different DST. Some places do not have any DST time, while some regions want to be on DST all year, instead of Standard time for half of the year. Some regions even have a different daylight time bias instead of the usual 1 hour.

Basically, you cannot determine DST based solely on a Timezone. You are going to have to manually adjust the DST for each "world clock" based on the location's DST rules (which can change at any time).

Here is some more reading on this: [https://forum.rainmeter.net/viewtopic.php?p=146550#p146552](https://forum.rainmeter.net/viewtopic.php?p=146550#p146552)

-Brian

**Comment by dixonte** [comment]
Hi @brianferguson,

I think perhaps I haven't properly explained the issue I'm having. Please allow me to try again in more detail.

The short version is this: I am not expecting Rainmeter to take daylight savings into account automatically. I expect to have to update the Timezone= option by hand when the clocks change in the region I am interested in. However, I find that I also have to change this value when the clocks change for ME, then again when they change for the target zone. I have to change each clock 4 times a year instead of 2.

I'm in the AEST (UTC+10) timezone. Recently we swapped over to AEDT (UTC+11), on the first Sunday of October.

I have a friend who lives in the US, in the CST/CDT timezone, and another that lives in Canada. Their clocks change on the first Sunday of November.

When the clocks changed in my timezone, all my desktop clocks became wrong, and I had to update them to match the time my friends reported. Even though the offset from UTC for those timezones had not changed.

From past experience I know that I will have to update the clocks again in November. However, this is the time I would expect to have to update the clocks, as the time offset relative to UTC has changed for those two friends.

Then in half a year, the local offset from UTC will change again as we swap back to AEST. At this point I will have to change my desktop clocks, even though the offset from UTC that they are meant to display has not changed.

And finally, when the UTC offset changes for my friends, I will have to update the clocks a 4th time. I would expect to have to update them at this time.

**Comment by brianferguson** [comment]
Are you using `DaylightSavingTime=0` ?  It is `1` by default.
[https://docs.rainmeter.net/manual/measures/time/#DaylightSavingTime](https://docs.rainmeter.net/manual/measures/time/#DaylightSavingTime)

-Brian

**Comment by dixonte** [comment]
@brianferguson I did not know about that option. That would explain the behaviour I've been seeing. Thanks, and sorry for the bother.

**Comment by brianferguson** [comment]
No bother at all!

-Brian
